### DISCUSSION TITLE
Gradle command failed

### DISCUSSION THREAD

**Discussion by ridmaur** [skiptools/skip#81] [Q&A] [answered]
The gradle command failed. Review the log for details and consult https://skip.tools/docs/faq for common solutions. Command: gradle -p ../Android launchDebug

**Comment by aabewhite** [comment]
Sorry for the trouble. Can you please post the full command you ran and the full output?

**Comment by ridmaur** [comment]
Just trying to run the sample Hello program following the instructions.

**Reply by aabewhite** [reply]
Is it working for you when you have the emulator running in Android Studio and then run the app via Xcode?

**Reply by ridmaur** [reply]
No that's exactly where it is failing. I do have Android Studio and emulator up and running and try to run the code from Xcode. With this error as a result.

**Reply by aabewhite** [reply]
Can you please post the full error from the Xcode build log?

**Reply by ridmaur** [reply]
I completely rerun the Hello Skippy example and it now worked. So problem solved.

One more question. Can I use skip.tools to convert an existing iOS app to Android (using Kotlin and JetCompose) and how would I do that. I can't seem to find that in the instructions.

I imagine something like:
- add the skip packages to the existing Xcode project
- run the code.

But I suspect there is more to it.

**Reply by aabewhite** [reply, marked as answer]
It's possible to add Skip to an existing project. Basically you have to mirror the way the template HelloSkip app is structured, and the way it includes Skip libraries and plugins in its Package.swift. The reason we don't document it well is that we don't generally recommend it.

Let's say that Skip is able to translate 95% of your project's code. That's still one in every 20 lines of code causing an error. That's a lot of errors at once, and we've found that it can be overwhelming. So even if Skip would save you 95% of the time of a manual Android port, developers tend to have a negative experience when applying it all at once to an existing project.

Instead I'd recommend creating a new Skip project - the docs cover this, including how to create multi-module projects - and then importing your code piecemeal. That way you can work with the tool on one piece at a time.
